### PR TITLE
python312Packages.nicegui: init at 2.5.0

### PR DESCRIPTION
**Pull request by fabaff**
Module to create web-based user interfaces

https://github.com/zauberzeug/nicegui/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352189`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>python311Packages.nicegui</li>
    <li>python311Packages.nicegui-highcharts</li>
    <li>python311Packages.nicegui-highcharts.dist</li>
    <li>python311Packages.nicegui.dist</li>
    <li>python311Packages.pyecharts</li>
    <li>python311Packages.pyecharts.dist</li>
    <li>python311Packages.pytest-selenium</li>
    <li>python311Packages.pytest-selenium.dist</li>
    <li>python311Packages.pytest-variables</li>
    <li>python311Packages.pytest-variables.dist</li>
    <li>python311Packages.vbuild</li>
    <li>python311Packages.vbuild.dist</li>
    <li>python312Packages.nicegui</li>
    <li>python312Packages.nicegui-highcharts</li>
    <li>python312Packages.nicegui-highcharts.dist</li>
    <li>python312Packages.nicegui.dist</li>
    <li>python312Packages.pyecharts</li>
    <li>python312Packages.pyecharts.dist</li>
    <li>python312Packages.pytest-selenium</li>
    <li>python312Packages.pytest-selenium.dist</li>
    <li>python312Packages.pytest-variables</li>
    <li>python312Packages.pytest-variables.dist</li>
    <li>python312Packages.vbuild</li>
    <li>python312Packages.vbuild.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/nicegui-highcharts/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  src = fetchFromGitHub {
+    owner = "zauberzeug";
+    repo = "nicegui-highcharts";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-r4X4faU7Nlq/FDbIYbTpvnC1w14XskpsNGtkEXtGrFo=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools>=30.3.0,<50" "setuptools"
```
Comment (natsukium) [inline pkgs/development/python-modules/nicegui-highcharts/default.nix:28]:
```suggestion
      --replace-fail "setuptools>=30.3.0,<50" ""
```
I don't think they actually use setuptools.
https://github.com/search?q=repo%3Azauberzeug%2Fnicegui-highcharts%20setuptools&type=code

File: pkgs/development/python-modules/nicegui/default.nix [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  src = fetchFromGitHub {
+    owner = "zauberzeug";
+    repo = "nicegui";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-oT191QVpvE5xszgBFt3o4A2hU50zmzPUywmAQuKZ5OE=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools>=30.3.0,<50" "setuptools"
```
Comment (natsukium) [inline pkgs/development/python-modules/nicegui/default.nix:57]:
```suggestion
      --replace-fail "setuptools>=30.3.0,<50" ""
```
same here
https://github.com/search?q=repo%3Azauberzeug%2Fnicegui%20setuptools&type=code

File: pkgs/development/python-modules/pytest-selenium/default.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    rev = "refs/tags/${version}";
+    hash = "sha256-fIyos73haqTAgp5WVvMwJswQAtXnsnUeXKjPweXLGRM=";
+  };
+
+  build-system = [
+    hatch-vcs
+    hatchling
+  ];
+
+  builInput = [ pytest ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pytest-selenium/default.nix:39]:
```suggestion
  buildInput = [ pytest ];
```
typo

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352189`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>python311Packages.nicegui</li>
    <li>python311Packages.nicegui-highcharts</li>
    <li>python311Packages.nicegui-highcharts.dist</li>
    <li>python311Packages.nicegui.dist</li>
    <li>python311Packages.pyecharts</li>
    <li>python311Packages.pyecharts.dist</li>
    <li>python311Packages.pytest-selenium</li>
    <li>python311Packages.pytest-selenium.dist</li>
    <li>python311Packages.pytest-variables</li>
    <li>python311Packages.pytest-variables.dist</li>
    <li>python311Packages.vbuild</li>
    <li>python311Packages.vbuild.dist</li>
    <li>python312Packages.nicegui</li>
    <li>python312Packages.nicegui-highcharts</li>
    <li>python312Packages.nicegui-highcharts.dist</li>
    <li>python312Packages.nicegui.dist</li>
    <li>python312Packages.pyecharts</li>
    <li>python312Packages.pyecharts.dist</li>
    <li>python312Packages.pytest-selenium</li>
    <li>python312Packages.pytest-selenium.dist</li>
    <li>python312Packages.pytest-variables</li>
    <li>python312Packages.pytest-variables.dist</li>
    <li>python312Packages.vbuild</li>
    <li>python312Packages.vbuild.dist</li>
  </ul>
</details>

**Pull request merged**